pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -311,19 +318,27 @@ func validateTaskRunResults(tr *v1beta1.TaskRun, resolvedTaskSpec *v1beta1.TaskS
 }
 
 // mismatchedTypesResults checks and returns all the mismatched types of emitted results against specified results.
-func mismatchedTypesResults(tr *v1beta1.TaskRun, specResults []v1beta1.TaskResult) map[string][]string {
-	neededTypes := make(map[string][]string)
-	providedTypes := make(map[string][]string)
-	// collect needed keys for object results
+func mismatchedTypesResults(tr *v1beta1.TaskRun, specResults []v1beta1.TaskResult) map[string]string {
+	neededTypes := make(map[string]string)
```

Comment:
I'm a bit confused why this was a map[string][]string earlier. Why would one result have multiple needed types?

Reply:
I think it was a mistake before. 😞